tests/all/gc.rs

```diff
@@ -237,9 +237,13 @@ fn drop_externref_via_table_set() -> anyhow::Result<()> {
 #[test]
 fn global_drops_externref() -> anyhow::Result<()> {
     test_engine(&Engine::default())?;
-    test_engine(&Engine::new(
-        Config::new().allocation_strategy(InstanceAllocationStrategy::pooling()),
-    )?)?;
+
+    // Avoid the pooling allocator when using QEMU for running tests (limited memory).
+    if std::env::var("WASMTIME_TEST_NO_HOG_MEMORY").is_err() {
```

Comment:
Could this check get moved to a helper function in `tests/all/gc.rs` to avoid duplicating the env var name too much?